pkg/controllers/namespace/namespace_controller.go

```diff
@@ -114,38 +127,168 @@ func getAccessMode(name string) v1alpha1.AccessMode {
+	if len(old.SubnetNames) != len(new.SubnetNames) {
+		return false
+	}
+	oldSubnets := make(map[string]struct{})
+	for _, subnet := range old.SubnetNames {
+		oldSubnets[subnet] = struct{}{}
+	}
+	for _, subnet := range new.SubnetNames {
+		if _, exists := oldSubnets[subnet]; !exists {
+			return false
+		}
+	}
+	return true
+}
+
+func listSubnetSet(ctx context.Context, k8sClient client.Client, ns string, label client.MatchingLabels) (*v1alpha1.SubnetSet, error) {
+	var oldObj *v1alpha1.SubnetSet
+	list := &v1alpha1.SubnetSetList{}
+	if err := k8sClient.List(ctx, list, label, client.InNamespace(ns)); err != nil {
+		return nil, err
+	}
+	if len(list.Items) > 0 {
+		log.Info("Default SubnetSet already exists", "label", label, "Namespace", ns)
+		oldObj = &list.Items[0]
+	}
+	return oldObj, nil
+}
+
+func listDefaultSubnetSet(ctx context.Context, k8sClient client.Client, ns string, subnetSetType string) (*v1alpha1.SubnetSet, error) {
+	networkSubnetSetNameMap := map[string]string{
+		types.DefaultPodNetwork: types.LabelDefaultPodSubnetSet,
+		types.DefaultVMNetwork:  types.LabelDefaultVMSubnetSet,
+	}
+	label := client.MatchingLabels{
+		types.LabelDefaultNetwork: subnetSetType,
+	}
+	oldObj, err := listSubnetSet(ctx, k8sClient, ns, label)
+	if err != nil {
+		return nil, err
+	}
+	// check the old formatted default subnetset
+	if oldObj == nil {
+		newType, _ := networkSubnetSetNameMap[subnetSetType]
+		label = client.MatchingLabels{
+			types.LabelDefaultSubnetSet: newType,
+		}
+		return listSubnetSet(ctx, k8sClient, ns, label)
+	}
+	return oldObj, nil
+}
+
+func (r *NamespaceReconciler) getDefaultSize() int {
+	defaultSubnetSize := util.MinSubnetSizeV90
+	if r.SubnetService.NSXClient.NSXCheckVersion(nsx.SubnetMinimalSize8) {
+		defaultSubnetSize = util.MinSubnetSizeV91
+	}
+	return defaultSubnetSize
+}
+
+// createDefaultSubnetSet only create default subnetset when sharedSubnet is empty for auto created vpc
+func (r *NamespaceReconciler) createDefaultSubnetSet(ctx context.Context, ns string, defaultSubnetSize int, sharedSubnet []v1alpha1.SharedSubnet, namespaceType NameSpaceType, networkStack v1alpha1.NetworkStackType) error {
+	if len(sharedSubnet) != 0 {
+		return nil
+	}
 	defaultSubnetSets := map[string]string{
-		types.DefaultVMSubnetSet:  types.LabelDefaultVMSubnetSet,
-		types.DefaultPodSubnetSet: types.LabelDefaultPodSubnetSet,
+		types.DefaultVMSubnetSet:  types.DefaultVMNetwork,
+		types.DefaultPodSubnetSet: types.DefaultPodNetwork,
 	}
+	if namespaceType == SystemNs {
+		// for system namespace, only create default vm subnetset
+		defaultSubnetSets = map[string]string{
+			types.DefaultVMSubnetSet: types.DefaultVMNetwork,
+		}
+	} else if namespaceType == SVServiceNs {
+		// for sv service namespace, only create default pod subnetset
+		defaultSubnetSets = map[string]string{
+			types.DefaultPodSubnetSet: types.DefaultPodNetwork,
+		}
+	}
+	vmSubnets, podSubnets := r.getVMPodSubnet(sharedSubnet)
```

Comment:
I feel getVMPodSubnet and related content can be removed as it's only used for log in this part?

Reply:
yes. removed.